rtos/source/ThisThread.cpp

```diff
@@ -251,4 +251,32 @@ const char *get_name()
 #endif
 }
 
+osPriority_t ThisThread::get_priority()
+{
+#if MBED_CONF_RTOS_PRESENT
+    osThreadId_t id = osThreadGetId();
+    if (id == nullptr) {
+        return osPriorityError;
+    }
+
+    return static_cast<osPriority_t>(osThreadGetPriority(id));
+#else
+    return static_cast<osPriority_t>(osPriorityError);
+#endif
+}
+
+osStatus_t ThisThread::suspend()
+{
+#if MBED_CONF_RTOS_PRESENT
+    osThreadId_t id = osThreadGetId();
+    if (id == nullptr) {
+        return osError;
+    }
+
+    return static_cast<osStatus_t>(osThreadSuspend(id));
+#else
+    return static_cast<osStatus_t>(osError);
+#endif
+}
```

Comment:
The changes to ThisThread and rtos/mbed_rtos_types.h should be done in their own commit with their justification clearly spelled out in the commit message description.

Reply:
If you referring to splitting bare-metal changes? then the bare-metal build will break because of the commit - https://github.com/ARMmbed/mbed-os/pull/12271/commits/be6f12bb19f3be1513c7ff94480811052e9a46af.

The OS wrapper which utilizes the `suspend and get_priority` method needs to be defined in the single commit both for bare-metal and RTOS builds.

Reply:
Do the changes in a commit on their own before any OS wrapper commits that require the changes. You are adding new functionality to Mbed OS C++ RTOS API. These could be upstreamed to master today.

Reply:
If I just split "Mbed OS C++ class" and "TF-M OS-Wrapper" changes separately, then "OS-Wrapper changes won't work on their own, as they have a dependency on the "Mbed OS C++ class" changes.

Reply:
Yup. You should also raise a separate PR to master for the C++ changes. I think @kjbracey-arm should review the changes for any potential issues.